sympy/polys/tests/test_ring_series.py

```diff
@@ -271,6 +273,17 @@ def test_atan():
         *x**3*y + EX((3*a**2 - 1)/(3*a**6 + 9*a**4 + 9*a**2 + 3))*x**3 + \
         EX(1/(a**2 + 1))*x**2*y - EX(a/(a**4 + 2*a**2 + 1))*x**2 + EX(1/(a**2 \
         + 1))*x + EX(atan(a))
+    # Complex numbers not in ring_series
+    R, x = ring('x', EX)
+    assert rs_atan(x + I,x,3) == EX(I/16)*x**2 + EX(S(1)/4)*x + EX(zoo + oo*I)
```

Comment:
`I` is a branch point of `atan` where no series expansion exists. I think that some kind of error should result.

Reply:
In ring_series,there is no check for singularities as well as no complex analysis has been done.The Polynomial ring is created with truncated values,so there is no check for branch points or branch cuts. Since with ``I`` or ``-I`` , ``atan`` gives values considering ``I`` to be a ``constant`` like ``pi``.

Reply:
Well,I want to implement complex analysis in ring_series by adding a function for singularity_check. I need leads to go on

Reply:
It seems to me that the two known branch points, `I` and `-I`, could be handled in the code. It should be necessary to check for singularities elsewhere.

Reply:
Alright then. Shall I add a patch for it?

Reply:
I think so. Perhaps `NotImplementedError`.

Reply:
I don't think passing ``x + I`` or ``x*I`` should raise ``NotImplementedError``. checking for ``rs_atan(I,x,5)`` results into ``AttributeError: 'ImaginaryUnit' object has no attribute 'ring' `` same as ``NotImplementedError``.So,shall I change the error message or try implementing the same.

Reply:
The values taken by `x + I` and `x*I` for `x = 0` are different. The former becomes `I` where `atan` is not defined while the latter is well defined at the origin. I think that your code should start by checking the center of the expansion. If it is `I` or `-I`, then `NotImplementedError` could be raised.

Reply:
okay got it.

Reply:
Not only ``I`` or ``-I`` , any multiples of ``I`` like ``2I`` or ``0*I`` should raise ``NotImplementedError``

Reply:
If the expression will be ``x+2I`` then putting ``x=-I`` will result into ``branch-point``. As the ``symbol`` ``x`` is not imposed as ``real`` while declaring, any ``Add`` object should raise error,rather ``Mul`` object can have their own series. Since as of here, ``x+I`` and ``x*I`` is a ``PolyElement`` object,it doesn't raise error.

Reply:
I think we need a ``_singularity_check`` function for ``PolyElement`` . It can't be checked likewise.

Reply:
Passing ``x+I`` shouldn't raise error as the expression is wrapped up by ``EX`` and manually putting values is not done.

Reply:
I have reverted the changes made earlier. I think it is good to go.

Reply:
I think that it is better to raise an expression than to return a meaningless expression like `EX(zoo + oo*I)`.

Reply:
@jksuom Actually, the problem is ``Add`` object should raise error while the expression ``x+I`` is evaluated,and it does. But using ``ring_series``, ``x+I`` becomes a ``PolyElement`` wrapped up with ``EX`` or ``QQ`` . A ``PolyElement`` object doesn't have ``is.real`` or ``is.imaginary`` attribute and it can't be checked if it was initially an ``Add`` object. ``x+I`` being a ``PolyElement`` causes difficulty and we don't want to use ``series`` in here and are using ``polys`` module

Reply:
It is not necessary to know that the polynomial `x + I` was initially an `Add` object. There is a function, `_get_constant_term`, for finding its constant term.

Reply:
ok,thanks

Reply:
For ``QQ`` and ``RR`` a ``TypeError`` is raised, but not for ``EX``. I have added the checks for ``EX(I)`` and ``EX(-I)``